nodebuilder/p2p/p2p_test.go

```diff
@@ -0,0 +1,216 @@
+package p2p
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p"
+	libhost "github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/metrics"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/p2p/host/autonat"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestP2PModule_methodsOnHost tests P2P Module methods on
+// the instance of Host.
+func TestP2PModule_methodsOnHost(t *testing.T) {
```

Comment:
[non-blocking] optional nit: i dont like the naming scheme of _methodsOnHost. I think TestP2PModule_Host would be better